modules/memfault-firmware-sdk/memfault_bt_metrics.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/atomic.h>
+
+#include <zephyr/logging/log.h>
+
+#include <memfault/metrics/metrics.h>
+
+#include "memfault_ncs_metrics.h"
+
+LOG_MODULE_DECLARE(memfault_ncs_metrics, CONFIG_MEMFAULT_NCS_LOG_LEVEL);
+
+static atomic_t connection_count = ATOMIC_INIT(0);
+
+#if CONFIG_MEMFAULT_NCS_STACK_METRICS
+static struct memfault_ncs_metrics_thread metrics_threads[] = {
+	{
+		.thread_name = "BT RX",
+		.key = MEMFAULT_METRICS_KEY(NcsBtRxUnusedStack)
+	},
+	{
+		.thread_name = "BT TX",
+		.key = MEMFAULT_METRICS_KEY(NcsBtTxUnusedStack)
+	}
+};
+#endif /* CONFIG_MEMFAULT_NCS_STACK_METRICS */
+
+
+void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	int err;
+	atomic_val_t count = atomic_get(&connection_count);
+
+	if (conn_err == 0) {
+		if (count == 0) {
+			err = memfault_metrics_heartbeat_timer_start(
+					MEMFAULT_METRICS_KEY(Ncs_BtConnectionTime));
+			if (err) {
+				LOG_WRN("Failed to start memfault Ncs_BtConnectionTime timer, "
+				"err: %d", err);
+			}
+		}
+
+		atomic_inc(&connection_count);
+	}
+}
+
+void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	int err;
+	atomic_val_t count = atomic_get(&connection_count);
+
+	__ASSERT_NO_MSG(count > 0);
+
+	atomic_dec(&connection_count);
```

Comment:
Detail: can this line be moved after the `if (count == 1) {` block for consistency with the `connected` callback?

Reply:
It can be moved. Done